prody/sequence/analysis.py

```diff
@@ -823,18 +823,28 @@ def buildMSA(sequences, title='Unknown', labels=None, **kwargs):
         try:
             max_len = 0
             for sequence in sequences:
-                if len(sequence) > max_len:
-                    max_len = len(sequence)
+                if isinstance(sequence, Atomic):
+                    if len(sequence.ca.copy()) > max_len:
+                        max_len = len(sequence.ca.copy())
+                elif isinstance(sequence, MSA):
+                    if len(sequence[0]) > max_len:
+                        max_len = len(sequence[0])
+                else:
+                    if len(sequence) > max_len:
+                        max_len = len(sequence)
 
             msa = []
             fetched_labels = []
             for i, sequence in enumerate(sequences):
                 if isinstance(sequence, Atomic):
-                    strseq = sequence.getSequence()
+                    strseq = sequence.ca.copy().getSequence()
```

Comment:
sequence.ca should also have getSequence. copy will need to cost extra memory.

Reply:
Ok true

Reply:
originally getSequence is exclusive to Chain type. But I have added it to all Atomic classes earlier so it should be safe.